cmd/infra/powervs/create.go

```diff
@@ -1554,10 +1560,15 @@ func (infra *Infra) createTransitGateway(ctx context.Context, options *CreateInf
 	}
 
 	log(options.InfraID).Info("Creating Transit Gateway ...")
+	// Getting respective vpc-region for powervs-region from region mapping.
+	vpcRegion, err := regionutils.VPCRegionForPowerVSRegion(options.Region)
+	if err != nil {
+		return nil, fmt.Errorf("respective vpc region not found for supplied powervs region:%s error:%w. Please use valid PowerVS and VPC regions", options.Region, err)
+	}
 	tg, _, err = tgapisv1.CreateTransitGatewayWithContext(ctx, &transitgatewayapisv1.CreateTransitGatewayOptions{
 		Location:      utilpointer.String(options.TransitGatewayLocation),
 		Name:          utilpointer.String(transitGatewayName),
-		Global:        utilpointer.Bool(true),
+		Global:        utilpointer.Bool((!(vpcRegion == options.VPCRegion)) || options.TransitGatewayGlobalRouting),
```

Comment:
```suggestion
		Global: globalRouting,
```

Reply:
Yes can be done but we need to update vendor package as well. Also I see this API is merged recently, It comes along with a lot of other changes as well. IMO, let's pull in latest package itself.

Reply:
Yeah we have updated the code there and removed the problem causing dependancy, you can pull the latest package and update the vendor.